episodes/chap5_python_client.md

```diff
@@ -35,75 +73,352 @@ server_api = "/api"
+]
 ```
 
+Typically, the `list()` method returns a paginated result.
+
+### Create
+Register a new *resource* at the server with `client.<resource>.create(...)`. For
+example, let's create a new organization:
+
 ```Python
-# Get an specific instance of the given '<resource>'.
-client.<resource>.get()
+client.organization.create(
+    name='new_organization',
+    address1='street 1',
+    address2='',
+    zipcode='1234AB',
+    country='NL',
+    domain='example.com',
+)
+```
+```output
+{
+    'nodes': '/api/node?organization_id=172',
+    'public_key': '',
+    'studies': '/api/study?organization_id=172',
+    'name': 'new_organization',
+    'tasks': '/api/task?init_org_id=172',
+    'address2': '',
+    'users': '/api/user?organization_id=172',
+    'domain': 'example.com',
+    'country': 'NL',
+    'zipcode': '1234AB',
+    'runs': '/api/run?organization_id=172',
+    'address1': 'street 1',
+    'id': 172,
+    'collaborations': '/api/collaboration?organization_id=172'
+}
 ```
 
-(\*) Please note that some resources do not provide all four operations. For example, it is not possible to create new rules.
-Also, note that the logged-in user may not be allowed to perform all operations.
+### Update
+Update an specific *resource*. For example:
 
-The parameters of the methods above vary based on the type of `<resource>` you are working with. To get these details, you can launch a Python interactive session that runs the client creation script above, and use the 'help' command with it. For example, `help(client.organization.list)` will show you that you can filter the list of organizations (among others) by name, country, and collaboration:
+```Python
+client.organization.update(id_, name='new_name')
+```
+```output
+{
+    'nodes': '/api/node?organization_id=173',
+    'public_key': '',
+    'studies': '/api/study?organization_id=173',
+    'name': 'new_name',
+    'tasks': '/api/task?init_org_id=173',
+    'address2': '',
+    'users': '/api/user?organization_id=173',
+    'domain': 'example.com',
+    'country': 'NL',
+    'zipcode': '1234AB',
+    'runs': '/api/run?organization_id=173',
+    'address1': 'street 1',
+    'id': 173,
+    'collaborations': '/api/collaboration?organization_id=173'
+}
+```
 
+### Delete
+To delete an specific instance of the given *resource* you can use
+`client.<resource>.delete(id_)` . Sometimes you need to also delete all the related
+instances of the given *resource* as well.
 ```Python
+client.organization.delete(1)
+```
+```output
+TODO: output, as this is not implemented in the current version
+```
+
+:::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: challenge
+
+### Collect collaboration details
+
+Before starting a task, you need to know the details of the collaboration you are working with. Use the Python client to get the details of the collaboration you have access to. Write down the **name** and and **ID** of each collaboration.
```

Comment:
Would it be an idea to point them to the option to use `fields=['id', 'name']`? I think that using this option makes this lesson much more enjoyable for the participants as they will not need to struggle through huge output dictionaries.

Reply:
Yes, good suggestion. Done